vtex-apps/listrak-pixel
# [LISTRAK-14] Fix shipping values sent to Listrak

#### What problem is this solving?

Listrak calculates order total as `order total = (itemtotal + taxtotal + shippingtotal + handlingtotal) - (merchandisediscount + nonmerchandisediscount)`. Shipping discounts are sent to Listrak in the `nonMerchandiseDiscount` field.

Assuming an order's shipping discount is distributed among all items in the orderForm, currently `nonMerchandiseDiscount` is set to the shipping discount applied to the first item only. This PR gets the total shipping discount applied to all items in the order and accounts for this discount by setting `shippingTotal` to the value before discounts.

#### How to test it?

Test in workspace [anna](https://anna--worldwidegolfqa.myvtex.com/). I created test order [78501687-01](https://worldwidegolfqa.myvtex.com/admin/orders/78501687-01). A free shipping promotion was applied to this order and the original shipping cost was $8.99. See this [log](https://vtex.splunkcloud.com/en-US/app/search/search?earliest=1663041600&latest=1663128000&q=search%20index%3Dio_vtex_logs%20app%3Dvtex.listrak-pixel*%20account%3Dworldwidegolfqa%20workspace%3Danna%20%22data.message%22%3D%22OrderHandler-createOrder%22%2078501687-01&display.page.search.mode=smart&dispatch.sample_ratio=1&sid=1663102670.21337406_0E315D9B-CDE1-4960-A650-072CF75459D7) for data sent to Listrak. `nonMerchandiseDiscount` and `shippingTotal` are both 8.99.
